### PR TITLE
Add test to Validate if duplicate metrics are intentional

### PR DESCRIPTION
**Pull request by mackjmr**
**Description:**
This test alerts a user if a metric is duplicated across receivers, and instructs the user to add said metric to an allow list if this is intentional.

If semantic conventions on the duplicate metric exist, validating is a matter of ensuring they both fulfill semantic intent. For validating duplicate metrics for which there is no semantic conventions, additional discussion is needed with the semantic conventions WG as mentioned in the issue. For the time being, it will be best judgment.

**Link to tracking Issue:** #26499

### CODE REVIEW

File: cmd/mdatagen/validate_test.go [hunk 1]
```diff
@@ -101,3 +105,48 @@ func TestValidate(t *testing.T) {
...
+			}
+		}
+		return nil
+	})
+	assert.NoError(t, err)
+
+	seen := make(map[string]string)
+	for receiver, metrics := range allMetrics {
+		for _, metricName := range metrics {
+			val, exists := seen[metricName]
```
Comment (songy23) [inline cmd/mdatagen/validate_test.go:133]: This looks a bit difficult to follow, I'd suggest
```
if val, exists := seen[metricName]; exists {
  // do the validation here
}
seen[metricName] = receiver
```

Comment (mackjmr) [reply cmd/mdatagen/validate_test.go]: Updated 👍

**Review by songy23 — APPROVED** [review]
LGTM

**Comment by mackjmr** [comment]
cc @dmitryax can you please take a look ?

File: cmd/mdatagen/validate_test.go [hunk 2]
```diff
@@ -101,3 +105,49 @@ func TestValidate(t *testing.T) {
...
+			assert.NoError(t, err)
+			if len(md.Metrics) > 0 {
+				for metricName := range md.Metrics {
+					allMetrics[md.Type] = append(allMetrics[md.Type], string(metricName))
+				}
+			}
+		}
+		return nil
+	})
+	assert.NoError(t, err)
```
Comment (mx-psi) [inline cmd/mdatagen/validate_test.go:128]:
```suggestion
	require.NoError(t, err)
```

File: cmd/mdatagen/validate_test.go [hunk 3]
```diff
@@ -101,3 +105,49 @@ func TestValidate(t *testing.T) {
...
+	allowedMetrics := map[string][]string{
+		"container.cpu.utilization": {"docker_stats", "kubeletstats"},
+		"container.memory.rss":      {"docker_stats", "kubeletstats"},
+		"container.uptime":          {"docker_stats", "kubeletstats"},
+	}
+	allMetrics := map[string][]string{}
+	err := filepath.Walk("../../receiver", func(path string, info fs.FileInfo, err error) error {
+		if info.Name() == "metadata.yaml" {
+			md, err := loadMetadata(path)
+			assert.NoError(t, err)
```
Comment (mx-psi) [inline cmd/mdatagen/validate_test.go:119]:
```suggestion
			require.NoError(t, err)
```

**Review by mx-psi — APPROVED** [review]

**Comment by mackjmr** [comment]
cc @dmitryax bumping this one again, let me know if you have any concerns with adding this ?

**Review by dmitryax — APPROVED** [review]

**Pull request merged**